src/Microsoft.DotNet.VersionTools/Dependencies/ProjectJsonUpdater.cs

```diff
@@ -29,33 +29,20 @@ public IEnumerable<BuildInfo> Update(IEnumerable<BuildInfo> buildInfos)
         {
             foreach (string projectJsonFile in ProjectJsonPaths)
             {
-                JObject projectRoot;
+                IEnumerable<BuildInfo> buildInfosUsed = null;
+
                 try
                 {
-                    projectRoot = ReadProject(projectJsonFile);
+                    FileUtils.ReplaceFileContents(projectJsonFile, contents => ReplaceAllDependencyVersions(contents, projectJsonFile, buildInfos, out buildInfosUsed));
                 }
                 catch (Exception e)
                 {
-                    Trace.TraceWarning($"Non-fatal exception occurred reading '{projectJsonFile}'. Skipping file. Exception: {e}. ");
+                    Trace.TraceWarning($"Non-fatal exception occurred reading or writing '{projectJsonFile}'. Skipping file. Exception: {e}. ");
```

Comment:
This is more than just reading and writing, on second thought--searching for the buildinfo and looking at json is in here too. I'm actually thinking that since the Maestro runs aren't typically monitored, we should have some way to put these warnings in the PR description so someone knows about them. (Probably something to file an issue for, though--this is fine by me for now.)